jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHolder.java

```diff
@@ -294,6 +294,22 @@ public void setForcedPath(String forcedPath)
         _forcedPath = forcedPath;
     }
 
+    private void copyClassServlet(ServletHolder holder) throws ServletException
+    {
+        if (holder.getClassName() != null)
+            this.setClassName(holder.getClassName());
+        // TODO adding this after setting the classname (or instead of) fixes the exceptions.
```

Comment:
I'm fine with setting the held class. All we need to do is to ensure that the loaded class is copied over, and thus the name of the class set etc.

Reply:
Ah, I got confused as I didn't realise the code to copy the instance was new.  
I'm still somewhat concerned by the duplicate configuration.

Reply:
@janbartel found that config is already handled